skbio/io/util.py

```diff
@@ -42,7 +47,17 @@ def _get_filehandle(filepath_or, *args, **kwargs):
     pass through.
     """
     if _is_string_or_bytes(filepath_or):
-        fh, own_fh = open(filepath_or, *args, **kwargs), True
+        if requests.compat.urlparse(filepath_or).scheme in {'http', 'https'}:
+            sess = CacheControl(requests.Session(),
+                                cache=FileCache(gettempdir()))
+            req = sess.get(filepath_or, **kwargs)
+
+            # if the response is not 200, an exception will be raised
+            req.raise_for_status()
+
+            fh, own_fh = BytesIO(req.content), True
```

Comment:
It may be necessary to pull down the full contents for things like xml or some binary like hdf5. I don't know if `h5py.File` will unroll a generator for instance, which I'm assuming would be the result of using `stream=True`? However, I think the hook in the registry like @ebolyen suggests should manage this.